x-pack/plugins/apm/server/routes/correlations/utils/get_duration_field.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  SPAN_DURATION,
+  TRANSACTION_DURATION,
+  TRANSACTION_DURATION_HISTOGRAM,
+} from '../../../../common/elasticsearch_fieldnames';
+import { ProcessorEvent } from '../../../../common/processor_event';
+
+export function getDurationField(eventType: ProcessorEvent) {
+  if (eventType === ProcessorEvent.metric) {
+    return TRANSACTION_DURATION_HISTOGRAM;
```

Comment:
I don't think `ProcessorEvent` is the right abstraction here anymore. We should be more specific, e.g.:

- `transaction_events`
- `span_events`
- `transaction_metrics`
- `service_destination_metrics`

Reply:
This new util function was just some code cleanup since we had repeated switching logic to for the duration field between the latency distribution chart and the correlations distribution charts. If the abstractions were to change, we would want to tease apart the correlations from the regular latency distribution chart (1st tab vs 2nd & 3rd tabs) into distinct functions for their own use cases. We could consider this tech debt and out of scope for this bug fix.

Reply:
The issue is that `eventType` in this case signals what _data_ we want to show, not what documents we should use to display it. Transaction latency data can be based on both ProcessorEvent.transaction and ProcessorEvent.metric, and similarly, exit span latency data can be based on both ProcessorEvent.span and ProcessorEvent.metric. I don't think we should conflate those concepts.